index.js

```diff
@@ -46,34 +47,119 @@ module.exports = {
     }
   },
 
+  flattenIcons: function (iconNames, subDir, srcDir) {
+    fs.readdirSync(srcDir).forEach((fileName) => {
+      var filePath = path.join(srcDir, fileName)
+      if (fs.lstatSync(filePath).isDirectory()) {
+        this.flattenIcons(iconNames, `${subDir}${subDir === '' ? '' : '/'}${fileName}`, filePath)
+      } else if (fileName.endsWith('.svg')) {
+        iconNames.push(`${subDir}/${fileName.substr(0, fileName.length - 4)}`)
+      }
+    })
+
+    return iconNames
+  },
+  
+  // Present purely to allow programmatic access to the icon packs and icon names (for demo purposes)
   treeForAddon: function (tree) {
```

Comment:
There is duplication in the detection of icon packs between `treeForAddon` and `treeForPublic` that I didn't have time to clean up (sorry) - be nice ;)